libraries/standard/mqtt/src/iot_mqtt_validate.c

```diff
@@ -39,6 +39,8 @@
 bool _IotMqtt_ValidateConnect( const IotMqttConnectInfo_t * pConnectInfo )
 {
     IOT_FUNCTION_ENTRY( bool, true );
+    uint16_t maxClientIdLength = ( uint16_t ) IOT_MQTT_SERVER_MAX_CLIENTID_LENGTH;
```

Comment:
Since client id is always a uint16_t, it should be defined as such

```c
#define IOT_MQTT_SERVER_MAX_CLIENTID_LENGTH ((uint16_t)128)
```

Reply:
Changed, also for the AWS IoT-specific one as well.